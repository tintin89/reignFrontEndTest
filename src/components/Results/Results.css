.results{   
     
    width: 100%;
    max-width: 1000px;
    margin-top: 2rem;    
    display: flex;   
    justify-content: space-between;   
    flex-wrap: wrap;
}

@media (max-width: 600px) {
    .results{       
        flex-flow: column;
        align-items: center;
    }
}