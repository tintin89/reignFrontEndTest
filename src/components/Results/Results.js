import React,{useEffect} from 'react';
import './Results.css';
import {useSelector} from 'react-redux'; 
import Element from './Element/Element';


const mapState=(myStateApp)=>({
    posts:myStateApp.posts
})

function Results() {
    const {posts} = useSelector(mapState);




    return (
        
        <div className="results">
           {
               
               posts.map((e,index)=>(
                   <Element key={index} {...e}/>
               ))
           }
        </div>
    )
}

export default Results
