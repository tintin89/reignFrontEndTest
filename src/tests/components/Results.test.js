import React from 'react';
import  {mount}  from "enzyme";
import Results from '../../components/Results/Results';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe('tests in  <Results/>',()=>{   
    const initState = {
        loading:false,        
    };
    let store = mockStore(initState);

    test('Must show correctly the component',()=>{       
        const wrapper = mount(
            <Provider store={store}>
             <Results/>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot()
    }) 
   
})