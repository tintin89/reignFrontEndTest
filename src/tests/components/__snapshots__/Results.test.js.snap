// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in  <Results/> Must show correctly the component 1`] = `
<div
  className="results"
/>
`;
